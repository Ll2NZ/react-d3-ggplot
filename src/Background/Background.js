import React from "react";
import { useContext } from "react";
import { ChartContext } from "../_context/";

function Background(props) {
  const context = useContext(ChartContext);

  return (
    <rect
      width={
        context.dimensions.width - context.dimensions.padding * 2
      }
      height={
        context.dimensions.height - context.dimensions.padding * 2
      }
      fill="rgb(232, 232, 232)"
      transform={`translate(${context.dimensions.padding}, ${
        context.dimensions.padding
      })`}
    />
  );
}

export default Background;
