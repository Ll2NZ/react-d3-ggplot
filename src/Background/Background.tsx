import React from "react";
import { useContext } from "react";
import { ChartContext } from "../_context/";
import { IContext } from "../_context/";

interface IProps {
  color?: string;
}

function Background(props: IProps) {
  const { dimensions } = useContext(ChartContext);

  return (
    <rect
      width={dimensions.width - dimensions.padding * 2}
      height={dimensions.height - dimensions.padding * 2}
      fill="#f1f1f1"
      transform={`translate(${dimensions.padding}, ${dimensions.padding})`}
    />
  );
}

export default Background;
