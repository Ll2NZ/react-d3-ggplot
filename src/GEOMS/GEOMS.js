import React            from "react";
import PropTypes        from "prop-types";
import { ChartContext } from "../_context/";

function GEOMS(props) {
        return(
            <ChartContext.Provider value={ props }>
                <svg width={ props.dimensions.width } height={ props.dimensions.height }>
                    { props.children }
                </svg>
            </ChartContext.Provider>
        );
}

GEOMS.propTypes = {
    dimensions: PropTypes.object,
    children: PropTypes.any
};

export default GEOMS;
