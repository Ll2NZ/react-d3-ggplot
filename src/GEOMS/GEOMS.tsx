import React, { ReactChildren, ReactComponentElement, ReactChild } from "react";
import { ChartContext, IContext } from "../_context/";

interface IProps extends IContext {
  children?: ReactChild[] | ReactChild;
}

function GEOMS(props: IProps) {
  if (!props.data || !props.aes || !props.dimensions) {
    throw new Error(`GEOMS must have the following properties: data, aes, and dimension`);
  }

  if (!props.children) {
    throw new Error("GEOMS expects at least one child components such as: <Rects/> or <Points/>");
  }

  return (
    <ChartContext.Provider value={props}>
      <svg width={props.dimensions.width} height={props.dimensions.height}>
        {props.children}
      </svg>
    </ChartContext.Provider>
  );
}

export default GEOMS;
