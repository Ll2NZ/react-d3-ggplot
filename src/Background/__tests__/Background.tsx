import React from "react";
import Background from "../Background";
import { GEOMS } from "../../GEOMS";
import { render } from "react-testing-library";
import { cleanup } from "react-testing-library";

afterEach(cleanup);

// here we only care about the dimensions property.
// <Background /> only needs dimensions object
const validState = {
  data: [],
  aes: [],
  dimensions: { width: 500, height: 300, padding: 50 }
};

test("<Background/> renders rect element", () => {
  // <Background/> is not intended to be used as a stand-alone componet, yet.
  // When using background, users must use it within <GEOMS/>
  const { container } = render(
    <GEOMS {...validState}>
      <Background />
    </GEOMS>
  );

  // If we change the code, the HTML structure should not change!
  expect(container.firstChild).toMatchSnapshot();
});
