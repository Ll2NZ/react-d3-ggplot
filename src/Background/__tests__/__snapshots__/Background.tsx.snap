// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background/> renders rect element 1`] = `
<svg
  height="300"
  width="500"
>
  <rect
    fill="#f1f1f1"
    height="200"
    transform="translate(50, 50)"
    width="400"
  />
</svg>
`;
